src/librustc_typeck/diagnostics.rs

```diff
@@ -2109,6 +2109,41 @@ E0380: r##"
 Default impls are only allowed for traits with no methods or associated items.
 For more information see the [opt-in builtin traits RFC](https://github.com/rust
 -lang/rfcs/blob/master/text/0019-opt-in-builtin-traits.md).
+"##,
+
+E0392: r##"
+This error indicates that a type parameter has been declared but not actually
+used.
+
+Here is an example that demonstrates the error:
+
+```
+enum Foo<T> {
+    Bar
+}
+```
+
+The first way to fix this error is by removing the type parameter, as
+shown below:
+
+```
+enum Foo {
+    Bar
+}
+```
+
+The second method is to actually make use of the type parameter:
+
+```
+enum Foo<T> {
+    Bar(T)
+}
+```
+
```

Comment:
Yeah, I figured that the more likely problem in this case was that the user simply forgot to use the parameter rather than wanting to use PhantomData - perhaps i should include some extra notes regarding PhantomData at the end.